connector/connector_ldap.go

```diff
@@ -217,11 +238,74 @@ func (c *LDAPConnector) TrustedEmailProvider() bool {
 	return c.trustedEmailProvider
 }
 
+type LDAPPool struct {
+	m          sync.Mutex
+	conns      map[*ldap.Conn]*ldap.Conn
```

Comment:
From small streams, mighty rivers can flow.

Your suggestion, `conns map[*ldap.Conn]struct{}`, looks neat and you can't do much better than zero. I'll use it.